examples/bvh_driver/bvh_driver.cpp

```diff
@@ -271,10 +271,8 @@ class CmdLineArgs
 
   ~CmdLineArgs()
   {
-    for (auto &p : _argv)
-    {
-      delete[] p;
-    }
+    for (int i = 0; i < _argc; ++i)
+      delete[] _argv[i];
```

Comment:
Isn't this the same?

Reply:
That's what I thought too, but it fixes the leak. I wonder if there is something about iterating on the data you remove at the same time.

Reply:
https://godbolt.org/z/QEVFBu

Reply:
So what does it tell us?

Reply:
Range-based for loop and the change you propose produce the same assembly output.
There is anything wrong with the old code (here)

Reply:
OK, you may tell me that, but that's not what I saw running on my workstation. When I ran the original variant, even without valgrind it complained about memory corruption, and refused to spit out Kokkos timers at the end of the program. After changing that, everything got fine.

Reply:
I'm also using `gcc`, not sure if that matters.

Reply:
Do you have a minimal reproducer?

Reply:
I don't, but you could rip the CmdLineArgs structure from bvh_driver into a standalone example.

Reply:
I did that, ran `-fsanitize=memory` and `valgrind` on it and couldn't detect anything.

Reply:
I just tried again directly running the test with `Valgrind` (after fixing the first bug) and didn't have any problems.